src/lpf2hub.ts

```diff
@@ -439,174 +453,117 @@ export class LPF2Hub extends Hub {
-                    const accelX = Math.round((data.readInt16LE(4) / 28571) * 2000);
-                    const accelY = Math.round((data.readInt16LE(6) / 28571) * 2000);
-                    const accelZ = Math.round((data.readInt16LE(8) / 28571) * 2000);
-                    /**
-                     * Emits when accelerometer detects movement. Measured in DPS - degrees per second.
-                     * @event LPF2Hub#accel
-                     * @param {string} port
-                     * @param {number} x
-                     * @param {number} y
-                     * @param {number} z
-                     */
-                    this.emit("accel", "ACCEL", accelX, accelY, accelZ);
-                    break;
-                }
-                case Consts.DeviceType.BOOST_TILT: {
-                    const tiltX = data.readInt8(4);
-                    const tiltY = data.readInt8(5);
-                    this._lastTiltX = tiltX;
-                    this._lastTiltY = tiltY;
-                    this.emit("tilt", port.id, this._lastTiltX, this._lastTiltY, this._lastTiltZ);
-                    break;
-                }
-                case Consts.DeviceType.POWERED_UP_REMOTE_BUTTON: {
-                    switch (data[4]) {
-                        case 0x01: {
-                            this.emit("button", port.id, Consts.ButtonState.UP);
-                            break;
-                        }
-                        case 0xff: {
-                            this.emit("button", port.id, Consts.ButtonState.DOWN);
-                            break;
-                        }
-                        case 0x7f: {
-                            this.emit("button", port.id, Consts.ButtonState.STOP);
-                            break;
-                        }
-                        case 0x00: {
-                            this.emit("button", port.id, Consts.ButtonState.RELEASED);
-                            break;
-                        }
-                    }
-                    break;
-                }
-                case Consts.DeviceType.DUPLO_TRAIN_BASE_COLOR: {
-                    if (data[4] <= 10) {
-                        this.emit("color", port.id, data[4]);
-                    }
-                    break;
-                }
-                case Consts.DeviceType.DUPLO_TRAIN_BASE_SPEEDOMETER: {
-                    /**
-                     * Emits on a speed change.
-                     * @event LPF2Hub#speed
-                     * @param {string} port
-                     * @param {number} speed
-                     */
-                    const speed = data.readInt16LE(4);
-                    this.emit("speed", port.id, speed);
-                    break;
+    private _emitSensorEvent(port: Port, mode: IPortMode, values: number[]) {
+        switch (mode.name) {
+            case "COLOR": {
+                this.emit("color", port.id, values[0]);
```

Comment:
Over all the breaking changes, it reads colors as a signed int so "no color" is -1. I think it make more sense but this is a very personal point of view.

Reply:
this was totaly wrong